src/domains/shielder/stores/getShielderIndexedDB.ts

```diff
@@ -73,17 +83,35 @@ const initDB = async (): Promise<IDBPDatabase<DBSchema>> => {
   });
 };
 
+const initDB = async (): Promise<IDBPDatabase<DBSchema>> => {
```

Comment:
You're calling this a couple of times  - don't you want to call this once at app startup and pass the promise around? Just like you do in the functions, but then you'd do it just once per the entire app instead of separately in every function.